### PR TITLE
feat: snackbar ui integration initiated

### PR DESCRIPTION
**Pull request by jhasuraj01**

### CODE REVIEW

File: packages/cysync-core/src/components/SnackBarManager.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+export const SnackBarManager = () => {
+  const snackBarState = useAppSelector(selectSnackBar);
+
+  if (!snackBarState.isOpen) {
+    return null;
+  }
+
+  // the below error will be fired in case props are not properly handled by redux state
+  if (!snackBarState.props) throw new Error('SnackBarProps is Undefined');
```
Comment (irshadCypherock) [inline packages/cysync-core/src/components/SnackBarManager.tsx]: Log and warning and return `null`

File: packages/cysync-core/src/store/snackBar/index.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  isOpen: false,
+  props: undefined,
+  snackBarId: undefined,
+};
+
+export const openSnackBar = createAsyncThunk<any, SnackBarProps>(
+  'snackBar/openSnackBar',
+  async () => {
+    await sleep(5000);
+  },
```
Comment (irshadCypherock) [inline packages/cysync-core/src/store/snackBar/index.ts]: The timeout for snackbar should be configurable

**Review by irshadCypherock — CHANGES_REQUESTED** [review]
- Open snackbar from all the required buttons. Example, when the user copies the transaction hash from history dialog.
- Run `pnpm lint; pnpm pretty`

File: packages/cysync-core/src/components/SnackBarManager.tsx (hunk 1 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/components/SnackBarManager.tsx]: updated

```ts
if (!snackBarState.props) {
    logger.warn('SnackBarProps is Undefined')
    return null;
  }
```

File: packages/cysync-core/src/store/snackBar/index.ts (hunk 2 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/store/snackBar/index.ts]: added timeout prop

File: packages/cysync-core/src/components/SnackBarManager.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import { useAppSelector } from '..';
+
+export const SnackBarManager = () => {
+  const snackBarState = useAppSelector(selectSnackBar);
+
+  if (!snackBarState.isOpen) {
+    return null;
+  }
+
+  // the below error will be fired in case props are not properly handled by redux state
```
Comment (irshadCypherock) [inline packages/cysync-core/src/components/SnackBarManager.tsx]: Remove comment

File: packages/ui/src/components/molecules/SnackBar.tsx [hunk 4]
```diff
@@ -36,28 +38,41 @@ const SnackBarButton = styled(Button)`
   margin: 4px 16px;
 `;
 
-interface SnackBarProps {
+type IconsType = 'check' | 'info';
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/SnackBar.tsx]:
```suggestion
export type SnackBarIconsType = 'check' | 'info';
```

File: packages/ui/src/components/molecules/SnackBar.tsx [hunk 5]
```diff
@@ -36,28 +38,41 @@ const SnackBarButton = styled(Button)`
...
-  imageSrc,
-  imageAlt,
-  buttonName,
+  icon = 'check',
+  buttonText,
 }) => (
   <SnackBarWrapper>
     <SnackBarPadding>
-      <Image src={imageSrc} alt={imageAlt} />
+      {/* <Image src={snackBarIcons[icon]} alt={icon} /> */}
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/SnackBar.tsx]: Remove comment

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**